Makefile

```diff
@@ -0,0 +1,112 @@
+# Run `make help` to display help
+
+# --- Global -------------------------------------------------------------------
+O = out
+COVERAGE = 0
+VERSION ?= $(shell git describe --tags --dirty  --always)
+REPO_ROOT = $(shell git rev-parse --show-toplevel)
+
+all: build test check-coverage lint  ## build, test, check coverage and lint
+	@if [ -e .git/rebase-merge ]; then git --no-pager log -1 --pretty='%h %s'; fi
+	@echo '$(COLOUR_GREEN)Success$(COLOUR_NORMAL)'
+
+ci: clean check-uptodate all  ## Full clean build and up-to-date checks as run on CI
+
+clean::  ## Remove generated files
+	-rm -rf $(O)
+
+.PHONY: all ci clean
+
+# --- Build --------------------------------------------------------------------
+GO_LDFLAGS = -X main.version=$(VERSION)
+CMDS = .
+
+build: | $(O)  ## Build reflect binaries
+	go build -o $(O) -ldflags='$(GO_LDFLAGS)' $(CMDS)
+
+install:  ## Build and install binaries in $GOBIN
+	go install -ldflags='$(GO_LDFLAGS)' $(CMDS)
+
+.PHONY: build install
+
+# --- Test ---------------------------------------------------------------------
+COVERFILE = $(O)/coverage.txt
+
+test: $(O) ## Run tests and generate a coverage file
+	go test -coverprofile=$(COVERFILE) ./...
+
+check-coverage: test  ## Check that test coverage meets the required level
+	@go tool cover -func=$(COVERFILE) | $(CHECK_COVERAGE) || $(FAIL_COVERAGE)
+
+cover: test  ## Show test coverage in your browser
+	go tool cover -html=$(COVERFILE)
+
+check-uptodate: protos
```

Comment:
lets move this up to where `ci` is - don't think it really belongs in the test section. It could be factored similar to how you did it for the README usage.

Reply:
ok.

Reply:
You should probably move the phony for `check-uptodate` too.

Reply:
done.